### PR TITLE
Add configurable thresholds for routing tables

### PR DESCRIPTION
**Pull request by jfim**
Add configurable thresholds for routing tables

Add configurable thresholds for determining whether or not to use the
large cluster routing table strategy. Added logging for the
configuration of all the routing table builders.

### CODE REVIEW

File: pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/KafkaLowLevelConsumerRoutingTableBuilder.java [hunk 1]
```diff
@@ -55,6 +55,7 @@ public void init(Configuration configuration) {
       final String targetServerCountPerQuery = configuration.getString("realtimeTargetServerCountPerQuery");
       try {
         TARGET_SERVER_COUNT_PER_QUERY = Integer.parseInt(targetServerCountPerQuery);
+        LOGGER.info("Using target server count of {}", TARGET_SERVER_COUNT_PER_QUERY);
```
Comment (mcvsubbu) [inline pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/KafkaLowLevelConsumerRoutingTableBuilder.java]: can u add in the log that this is for realtime nservers? A more descriptive log may also help.

File: pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/LargeClusterRoutingTableBuilder.java [hunk 2]
```diff
@@ -156,6 +156,7 @@ public void init(Configuration configuration) {
       final String targetServerCountPerQuery = configuration.getString("offlineTargetServerCountPerQuery");
       try {
         TARGET_SERVER_COUNT_PER_QUERY = Integer.parseInt(targetServerCountPerQuery);
+        LOGGER.info("Using target server count of {}", TARGET_SERVER_COUNT_PER_QUERY);
```
Comment (mcvsubbu) [inline pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/LargeClusterRoutingTableBuilder.java]: same. pls add that this is for offline target servers.

File: pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/KafkaLowLevelConsumerRoutingTableBuilder.java [hunk 3]
```diff
@@ -55,6 +55,7 @@ public void init(Configuration configuration) {
...
       } catch (Exception e) {
         LOGGER.warn(
             "Could not get the target server count per query from configuration value {}, keeping default value {}",
```
Comment (mcvsubbu) [inline pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/KafkaLowLevelConsumerRoutingTableBuilder.java]: canu modify this log as well to include the info that this is for realtime routing tables?

File: pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/LargeClusterRoutingTableBuilder.java [hunk 4]
```diff
@@ -156,6 +156,7 @@ public void init(Configuration configuration) {
...
       } catch (Exception e) {
         LOGGER.warn(
             "Could not get the target server count per query from configuration value {}, keeping default value {}",
```
Comment (mcvsubbu) [inline pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/LargeClusterRoutingTableBuilder.java]: and this one to include that it is for offline tables

File: pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/KafkaLowLevelConsumerRoutingTableBuilder.java (hunk 1 above)
Comment (jfim) [inline pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/KafkaLowLevelConsumerRoutingTableBuilder.java]: It has the class name:
```
INFO [KafkaLowLevelConsumerRoutingTableBuilder] [main] [pinot-broker] [] Using default value for target server count of 8
```

**Pull request merged**